build/run.sh

```diff
@@ -2,6 +2,16 @@
 
 set -e
 
-distro=$(cat /etc/os-release | grep -E "^ID=" | cut -d= -f2 | sed -e 's/"//g')
+# Determine the distribution ID
+distro=$(grep -E "^ID=" /etc/os-release | cut -d= -f2 | tr -d '"')
 
-"$(dirname "$0")/${distro}/$1".sh
\ No newline at end of file
+# Construct the script path
+script_path="$(dirname "$0")/${distro}/$1.sh"
+
+# Check if the script exists and execute it, or exit with an error
+if [ -x "$script_path" ]; then
+    "$script_path"
+else
+    echo "Error: Script $script_path not found or not executable."
+    exit 1
+fi
```

Comment:
I'm not sure how this improves the interpretation of the failure. It was pretty obvious failure before if the script wasn't there.

Reply:
Your corrected. Reverted back.